### ISSUE TITLE
Prisma migration fails on both update and clean install

### ISSUE THREAD

**Issue by mguth** [umami-software/umami#1665]
Hi!

I have recently tried both an update and a fresh install of umami (v1.39.5) on my shared server (nginx, Node 16, MariaDB 10.3).

In all cases (this example is for a fresh empty DB), they failed during the migration step with the following error (full log attached):

``` 
Migration name: 02_add_event_data

Database error code: 1172

Database error:
Result consisted of more than one row

Please check the query number 16 from the migration file.


✗ Command failed: prisma migrate deploy
Error: P3018
```
[yarn-build.log](https://github.com/umami-software/umami/files/10039843/yarn-build.log)

**Comment by briancao** [comment]
For MariaDB, you'll need to be on version 10.5.2 or higher.

**Comment by chreniuc** [comment]
@briancao I'm using mariadb 10.6 and still getting the same error(on clean install latest version):

```bash
Prisma schema loaded from prisma/schema.prisma
Datasource "db": MySQL database "umami" at "localhost:3306"
Migration 01_init marked as applied.

Error: P3018

A migration failed to apply. New migrations cannot be applied before the error is recovered from. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve

Migration name: 02_add_event_data

Database error code: 1172

Database error:
Result consisted of more than one row

Please check the query number 16 from the migration file.


✗ Command failed: prisma migrate deploy
Error: P3018

A migration failed to apply. New migrations cannot be applied before the error is recovered from. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve

Migration name: 02_add_event_data

Database error code: 1172

Database error:
Result consisted of more than one row

Please check the query number 16 from the migration file.



error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
ERROR: "check-db" exited with 1.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
/app $ 

```

**Comment by chreniuc** [comment]
This was a strange error, it fails because I had two databases for umami(two different names) in the same mariadb server. Why is it influenced by this? It fails when running the query from [here](https://github.com/umami-software/umami/blob/master/db/mysql/migrations/02_add_event_data/migration.sql#L84):

```sql
CALL UmamiRenameIndexIfExists('account', 'username', 'account_username_key');
```

Which calls:

```sql
SELECT
        1
    INTO @indexExists FROM
        INFORMATION_SCHEMA.STATISTICS
    WHERE
        TABLE_NAME = @tableName
            AND INDEX_NAME = @currentIndexName;
```

Which means that if in your db server you have another database which has a table called: `account` and an index called `username`, it will fail. I don't think this is ok..

That query should also take into consideration the db name, which means that the query should be:

```sql
SELECT
        1
    INTO @indexExists FROM
        INFORMATION_SCHEMA.STATISTICS
    WHERE
        TABLE_NAME = @tableName
            AND INDEX_NAME = @currentIndexName
           AND TABLE_SCHEMA = 'umami'; # or whatever name it is
```

**Comment by chreniuc** [comment]
That was because I was using a super user in mariadb, you need a dedicated user only for umami: https://github.com/umami-software/umami/issues/1753

**Comment by manishh** [comment]
I'm using a dedicated user for MariaDB/Umami, but stall facing this error :(

Any workaround this?

**Comment by ayewo** [comment]
@chreniuc your comment just saved me some valuable time as I had been head scratching as to why the migration was failing for a fresh database. My setup is identical to yours i.e. two differently named umami databases except that I'm using MySQL 8.0.22 on Ubuntu.

@briancao would you be open to accepting a PR to fix the underlying issue that @chreniuc explained [above](https://github.com/umami-software/umami/issues/1665#issuecomment-1397140327) once and for all?

**Comment by briancao** [comment]
@chreniuc Thanks for identifying the issue/fix.  

@ayewo Editing and altering migration history can potentially cause a lot of issues, so we'll be fixing and testing this internally.  So no need for a PR (although appreciated).

**Comment by franciscao633** [comment]
@chreniuc Appreciate the help with this. This has been fixed in v1.40.0
